zul/src/org/zkoss/zul/impl/NumberInputElement.java

```diff
@@ -122,8 +122,13 @@ public void setLocale(Locale locale) {
 		if (!Objects.equals(_locale, locale)) {
 			_locale = locale;
 
-			if (getFormat() == null)
-				setFormat(getDefaultFormat());
+			if (getFormat() == null) {
+				if ("en_IN".equals(locale.toString())) {
```

Comment:
Hard-coded. What if we add another format in the future?

Reply:
ok